src/libraries/Common/tests/System/Net/Configuration.Http.cs

```diff
@@ -12,11 +12,14 @@ public static partial class Http
         {
             private static readonly string DefaultHttp2AzureServer = "corefx-net-http2.azurewebsites.net";
 
+            // for the local server hosted in XHarness we are passing also port as part of the environment variables, because it's bound to random port number
             public static string Host => GetValue("DOTNET_TEST_HTTPHOST", DefaultAzureServer);
-
             public static string SecureHost => GetValue("DOTNET_TEST_SECUREHTTPHOST", DefaultAzureServer);
-
             public static string Http2Host => GetValue("DOTNET_TEST_HTTP2HOST", DefaultHttp2AzureServer);
+            public static string Http2SecureHost => GetValue("DOTNET_TEST_HTTP2SECUREHOST", DefaultHttp2AzureServer);
+
+            public static int Port = GetPortValue("DOTNET_TEST_HTTPHOST", 80);
+            public static int SecurePort = GetPortValue("DOTNET_TEST_SECUREHTTPHOST", 443);
```

Comment:
`DOTNET_TEST_HTTPHOST` -> `DOTNET_TEST_HTTP2HOST`

It would be better to extract the port number from `Http2Host`, and `Http2SecureHost` strings instead. And name these `Http2Port`, `Http2SecurePort`?

Reply:
Oh, the echo server is `httpHost`, and `http2Host`. This is fine then, my bad. You could extract the value from `Host`, and `SecureHost`, but that's a nit.

Reply:
The usage is together with `SecureHost` like `withPort ? Configuration.Http.SecureHost + ":" + Configuration.Http.SecurePort`